base/scripts/code.game.cs

```diff
@@ -1078,3 +1081,43 @@ function remoteSetArmor(%player, %armorType)
   }
   
 }
+
+function Game::BodyBlockCheck()
+{
+    
+    if(!$BodyBlock::Enabled) { return; }
+    
+    //if we in a pause, back out and try again in 1 second
+    if ($freezedata::actice == 1) {
+        schedule("Game::BodyBlockCheck();", 1);
+        return;
+    }
+    
+    //only look for player loc if the match has begun
+    if ($matchStarted) {
+        
+        //cycle through clients checking to see if their distance falls within the flag zone
+        for(%cl = Client::getFirst(); %cl != -1; %cl = Client::getNext(%cl)) {
+            %clTeam = Client::getTeam(%cl);
+            if (%clTeam == 0 || %clTeam == 1) {
+                %otherTeam = (%clTeam + 1) % 2; //if 0 returns 1, if 1 returns 0.
+            
+                //check to see if we are in the process of calculating a BB first
+                if (!$BodyBlock::Calculate[%cl]) {
+                
+                    //find speed
+                    $BodyBlock::Speed[%cl] = Game::BodyBlockDistance(%cl, %otherTeam);
+                }
+            }
+        }
+    }
+    schedule("Game::BodyBlockCheck();", 1);
+}
+
+function Game::BodyBlockDistance(%cl, %otherTeam)

```

Comment:
nit: probably a better name to be like `getPlayerSpeed`, also the `%otherTeam` parameter doesn't look used:

```suggestion
function Game::getPlayerSpeed(%cl)
```

the `Vector::getDistance` method is essentially taking the magnitude of the vector so the name is misleading

Reply:
initially this function was created when I had flag play incorporated, so i needed distance from the opposing teams flag, but i removed that when you suggested all body block's should be account for not ones solely near a flag. the name of the function remained untouched

I left %otherTeam in there for the same reason as above, and depending on how bodyblock tracking goes it might be used in the future? still undecided but what to leave it for now

Reply:
thanks for the explanation for the historical reason why the function is named that way. i would say that given that the way things are calculated have changed since then it feels appropriate to update both the name and the function signature

this is a shared code base, so the time for draft work is in your personal branch, but when merged upstream things should be pristine and ready to go

Reply:
done